dapps/assets/src/components/contacts/AddContact.ts

```diff
@@ -0,0 +1,130 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+
+// evan.network imports
+import { EvanComponent, EvanForm } from '@evan.network/ui-vue-core';
+import { ContactsService } from './ContactsService';
+
+@Component
+export default class AddContactComponent extends mixins(EvanComponent) {
+  contactService: ContactsService;
+
+  addressbook;
+
+  idOrEmailValidation = '';
+  idOrEmail = null;
+
+  accountId = null;
+  alias = null;
+  email = null;
+  emailInvite = null;
+  fromAlias = window.localStorage.getItem('evan-alias');
+  msgBody = `${this.$t('_assets.contacts.message-prefill')}${this.fromAlias}`;
+  msgTitle = this.$t('_assets.contacts.subject-prefill');
+
+  async created() {
+    this.initState();
+    const runtime = this.getRuntime();
+    this.contactService = new ContactsService(runtime);
+    this.addressbook = await runtime.profile.getAddressBook();
+  }
+
+  private initState() {
```

Comment:
looks bit ugly because of the code duplications but I guess there is no other way around to have the vars reactive ...

Reply:
Nope, there isn't a better solution, unless we hack some different logic into it to reset component state dynamically.

Reply:
As i mentioned before, we can just set initial values to null, so we avoid duplicated code, but keep reactivity